lib/internal/socketaddress.js

```diff
@@ -138,6 +140,32 @@ class SocketAddress {
       flowlabel: this.flowlabel,
     };
   }
+
+  /**
+   * Parse an "${ip}:${port}" formatted string into a SocketAddress.
+   * Returns undefined if the input cannot be successfully parsed.
+   * @param {string} input
+   */
+  static parse(input) {
+    validateString(input, 'input');
+    try {
+      const {
+        hostname: address,
+        port,
+      } = URL.parse(`http://${input}`);
```

Comment:
Since we are using `URL.parse()` we can just avoid having the try/catch and update line 156 with `url?.hostname.startsWith('[')`

Reply:
the `try { } catch { }` is mostly there for the `new SocketAddress(...)` constructor, not `URL.parse(...)`